frontend/readhistory.lua

```diff
@@ -295,6 +281,14 @@ function ReadHistory:addItem(file, ts, no_flash)
     end
 end
 
+--- Updates last book access time on closing the document.
+function ReadHistory:updateLastBookTime()
+    local now = os.time()
+    self.hist[1].time = now
+    self.hist[1].mandatory = datetime.secondsToDateTime(now)
+    self:_flush()
```

Comment:
Just asking, probably not worth complicating things :):
When switching documents, we will write the history to disk when closing the previous document, and also when loading the new document. No way to avoid one of them ? (ie. updating in memory self.hist, but avoiding the _flush() as not essential if it's still the last document?)

Reply:
Maybe the `tearing_down` flag can help, I'll look at it.